crates/term/lib.rs

```diff
@@ -378,12 +381,12 @@ impl<D: Decorator + Send + Sync> StreamFormat for Format<D> {
 
 const TIMESTAMP_FORMAT: &'static str = "%b %d %H:%M:%S%.3f";
 
-fn timestamp_local() -> String {
-    chrono::Local::now().format(TIMESTAMP_FORMAT).to_string()
+fn timestamp_local(sink: TimestampSink) -> io::Result<()> {
+    sink(format_args!("{}", chrono::Local::now().format(TIMESTAMP_FORMAT)))
```

Comment:
Taking timestamp in chrono is terribly slow right now BTW, and I've filled an issue about it: https://github.com/lifthrasiir/rust-chrono/issues/94 . I'm hopping that chrono devs will provide with faster APIs. Worst case I'd fork chrono and write the methods myself. :]

Note that `chrono::Local::now().format(TIMESTAMP_FORMAT)` is returning a newly allocated String.

```
fn fmt_rfc3339<W : io::Write>(&self, io : W) -> io::Result<()>
```

so chrono has an allocation free API too.

With the current version of the PR the `format_args!("{}", val )` part will require `val` to be something that another function can return - like `String`, forcing allocation anyway or some gimmicks around reusing buffers.

I have spent quite a bit of time optimizing slog  and I might be wrong, but AFAIK the best way to have fast formatting of anything is to pass the formatter the `io::Write` or `fmt::Write` and let it write to it directly.

If the timestamp fn would be:

```
Box<(Fn(&mut io::Write) -> io::Result<()>) + Send + Sync>;
```

(I'm not sure if these additional `( )` are needed, and I wonder if they might be since.

Then `timstamp_local` would be just:

```
fn timestamp_local(&mut io::Write) -> io::Result<()> {
    write!("{}", chrono::Local::now().format(DEFAULT_TIMESTAMP_FORMAT))
}
```

which seems OK to me.